JavaFiddle-ejb/src/main/java/ru/javafiddle/core/ejb/UserBean.java

```diff
@@ -1,180 +1,152 @@
 package ru.javafiddle.core.ejb;
 
-import java.util.LinkedList;
-import java.util.List;
+import ru.javafiddle.jpa.entity.Status;
+import ru.javafiddle.jpa.entity.User;
+import ru.javafiddle.jpa.entity.UserGroup;
+
 import javax.annotation.Resource;
 import javax.ejb.SessionContext;
 import javax.ejb.Stateless;
-import javax.faces.bean.ManagedBean;
+import javax.ejb.TransactionAttribute;
+import javax.ejb.TransactionAttributeType;
 import javax.persistence.EntityManager;
 import javax.persistence.NoResultException;
 import javax.persistence.PersistenceContext;
 import javax.persistence.TypedQuery;
 
-import ru.javafiddle.jpa.entity.*;
-
 import java.text.DateFormat;
 import java.text.SimpleDateFormat;
+
 import java.util.Date;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.Iterator;
+import java.util.List;
+
 /**
  *
  * @author mac
  */
 @Stateless
 public class UserBean {
 
-    private static final String DEFAULT_USER_STATUS = "registered";
-
     @Resource
     private SessionContext ctx;
 
-    @PersistenceContext(name = "JFPersistenceUnit")
-    private EntityManager em;
+    private static final Logger logger =
+            Logger.getLogger(UserBean.class.getName());
 
-    public UserBean() {
-    }
+    private static final int DEFAULT_USER_STATUS = 1;
 
-    public User register(String firstName, String lastName, String nickname, String email, String passwordHash) {
+    @PersistenceContext(name = "JFPersistenceUnit")
+    EntityManager em;
 
-        User user = new User();
-        Status status = (Status)em.createQuery("SELECT s FROM Status s WHERE s.statusName =:status")
-                .setParameter("status", DEFAULT_USER_STATUS)
-                .getSingleResult();
+    public UserBean() {
 
+    }
 
+    @TransactionAttribute(TransactionAttributeType.REQUIRES_NEW)
+    public User register(User user) {
+        //search for registered class
+        Status st = em.find(Status.class, DEFAULT_USER_STATUS);
+        if (checkCorrectnessOfNick(user.getNickName())) {
+            user.setStatus(st);
 
-        user.setFirstName(firstName);
-        user.setLastName(lastName);
-        user.setNickName(nickname);
-        user.setEmail(email);
-        user.setPasswordHash(passwordHash);
+            DateFormat dateFormat = new SimpleDateFormat("yyyy/MM/dd HH:mm:ss");
```

Comment:
Consider initializing it on a class level (e.g. private final field or even static...)